libqtile/widget/base.py

```diff
@@ -369,6 +370,35 @@ def mouse_enter(self, x, y):
     def mouse_leave(self, x, y):
         pass
 
+    def _draw_with_mirrors(self):
+        self._old_draw()
+        for mirror in self.mirrors:
+            if not mirror.configured:
+                continue
+
+            if mirror.length_type == bar.CALCULATED:
+                mirror.bar.draw()
+            else:
+                mirror.draw()
+
+    def add_mirror(self, widget: _Widget):
+        if not self.mirrors:
+            self._old_draw = self.draw
+            self.draw = self._draw_with_mirrors  # type: ignore
+
+        self.mirrors.add(widget)
+        self.drawer.add_mirror(widget.drawer)
+
+    def remove_mirror(self, widget: _Widget):
+        try:
+            self.mirrors.remove(widget)
+            self.drawer.remove_mirror(widget.drawer)
+        except KeyError:
+            pass
+
+        if not self.mirrors and hasattr(self, "_old_draw"):
+            self.draw = self._old_draw  # type: ignore
```

Comment:
We might be able to get away with not saving `self._old_draw` and getting it from `self.__class__.draw`.

(2 mins later after some testing...) it looks like restoring it with `self.__class__.draw` causes a `missing 1 required positional argument: 'self'`, but `types.MethodType` can fix this: `self.draw = types.MethodType(self.__class__.draw, self)`. See: https://stackoverflow.com/questions/47797661/python-types-methodtype

But actually that's also unnecessary. Methods are looked up in the MRO when needed using the class definitions, so it's as straightforward as doing `del self.draw`, then later references to `self.draw` should get the original method :D

Reply:
I'd looked at `MethodType` and wanted to ignore it. `del self.draw` is neat!